crates/fluvio-version-manager/src/command/switch/version_directory.rs

```diff
@@ -0,0 +1,224 @@
+
+use anyhow::Result;
+
+use crate::common::manifest::{PACKAGE_SET_MANIFEST_FILENAME, VersionManifest};
+use crate::common::workdir::fluvio_binaries_path;
+
+/// Represents the contents of a version directory (`~/.fvm/versions/<version>`)
+/// where binaries and the manifest are stored.
+///
+/// The `VersionDirectory` represents the contents of a version directory
+/// in 2 parts:
+///
+/// - `contents`: Every file in the directory except for the manifest
+/// - `manifest`: The manifest file
+pub struct VersionDirectory {
+    pub contents: Vec<PathBuf>,
+    pub manifest: VersionManifest,
+}
+
+impl VersionDirectory {
+    /// Opens a directory and returns a `VersionDirectory` build from scanning it
+    pub fn open(path: PathBuf) -> Result<Self> {
+        let mut contents: Vec<PathBuf> = Vec::new();
+        let mut manifest: Option<VersionManifest> = None;
+
+        for entry in read_dir(path)? {
+            let entry = entry?;
+
+            if entry.metadata()?.is_dir() {
+                continue;
+            }
+
+            let entry_path = entry.path();
+
+            if entry_path.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry_path.display()
+            ))? == PACKAGE_SET_MANIFEST_FILENAME
+            {
+                manifest = Some(VersionManifest::open(&entry_path)?);
+            } else {
+                contents.push(entry_path);
+            }
+        }
+
+        let Some(manifest) = manifest else {
+            return Err(anyhow::anyhow!(
+                "Failed to find manifest file in version directory"
+            ));
+        };
+
+        Ok(Self { contents, manifest })
+    }
+
+    /// Sets this version as the active Fluvio Version
+    pub fn set_active(&self) -> Result<()> {
+        // Verify `~/.fluvio/bin` exists and create it if it doesn't
+        let fluvio_bin_dir = fluvio_binaries_path()?;
+
+        if !fluvio_bin_dir.exists() {
+            tracing::info!(?fluvio_bin_dir, "Creating fluvio binaries directory");
+            create_dir_all(&fluvio_bin_dir)?;
+        }
+
+        for entry in &self.contents {
+            let filename = entry.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry.display()
+            ))?;
+            let target_path = fluvio_bin_dir.join(filename);
+
+            copy(entry, &target_path)?;
+            tracing::info!(?target_path, "Copied binary");
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::{remove_file, remove_dir_all};
+    use std::path::Path;
+
+    use anyhow::Result;
+
+    use tempfile::TempDir;
+
+    use fluvio_hub_util::sha256_digest;
+
+    use super::*;
+
+    /// Binaries included in a Fluvio Version. These are mapped to the ones in
+    /// the `fixtures/version` directory
+    const EXPECTED_BINARIES: [&str; 5] = ["cdk", "fluvio", "fluvio-cloud", "fluvio-run", "smdk"];
```

Comment:
This is a pretty file to check into git.  Instead, create a tiny executable file (a bash file would also work). As long as they have executable and checksum can be created